parsely/src/main/java/com/parsely/parselyandroid/ParselyMetadata.kt

```diff
@@ -21,13 +21,13 @@ open class ParselyMetadata
  * @param title    The title of the content.
  * @param publicationDateMilliseconds  The date this piece of content was published.
  */(
-    private val authors: List<String>?,
-    @JvmField internal val link: String?,
-    private val section: String?,
-    private val tags: List<String>?,
-    private val thumbUrl: String?,
-    private val title: String?,
-    private val publicationDateMilliseconds: Long?
+    private val authors: List<String>? = null,
+    @JvmField internal val link: String? = null,
+    private val section: String? = null,
+    private val tags: List<String>? = null,
+    private val thumbUrl: String? = null,
+    private val title: String? = null,
+    private val publicationDateMilliseconds: Long? = null
```

Comment:
Instead of this, I was exploring introducing a builder pattern, so Java clients won't have to provide `null` in place of nullable fields.

I decided not doing it, as it'd deviate more from the original implementation, making addressing the breaking change more difficult. Furthermore, it'd be much different than iOS API. So I think, the default values for arguments will be just fine - making the API nicer for Kotlin users and not changing it for Java users.

Reply:
As far as I know, using Builder pattern would mean that we're adding to the API not changing it. We'd still have the `ParselyMetadata` without any breaking changes - but there would also be a `ParselyMetadataBuilder` which returns `ParselyMetadata` when we call `.build()` on it.

That's all to say that, if Builder pattern is something you'd like to utilize, I don't think it conflicts with the current implementation.

Reply:
That's true - thanks. I've tunneled vision on constructor OR builder, but there's no reason to not introduce builder separately from the constructor. Still, for now, I'd leave this as it is and keep the builder for possible enhancement in the future.